packages/wrangler/pages/functions/filepath-routing.ts

```diff
@@ -205,7 +203,9 @@ async function forEachFile<T>(
   const returnValues: T[] = [];
 
   while (searchPaths.length) {
-    const cwd = searchPaths.shift();
+    // The `searchPaths.length` check above ensures that `searchPaths.shift()` is defined
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const cwd = searchPaths.shift()!;
```

Comment:
It is a shame that TS cannot infer this.

Reply:
could include in the `while` check a `&& Array.isArray(searchPaths)` it could be not inferring because `.length` can also be on a string.

Reply:
or alternatively 
```suggestion
    const cwd = Array.isArray(searchPaths) && searchPaths.shift()!;
```

Reply:
If we run across a lot of these we can make a custom type-guard, this example isn't inferring anything from inputs or using generics.
```typescript
function isNonEmptyArrStrings(value: any): boolean {
    return Array.isArray(value) && value.length && value.every(item => typeof item === "string");
}
```

Reply:
I'm going to add the guard because that is quite cool